handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -638,9 +638,7 @@ public final SSLEngineResult wrap(
                         bytesProduced += bioLengthBefore - pendingNow;
                         bioLengthBefore = pendingNow;
 
-                        if (bytesConsumed == MAX_PLAINTEXT_LENGTH || bytesProduced == dst.remaining()) {
-                            return newResultMayFinishHandshake(status, bytesConsumed, bytesProduced);
-                        }
```

Comment:
If we consume multiple packets we must be more careful about how much overhead we account for. This is done in PR https://github.com/netty/netty/pull/6365